packages/aws-cdk/lib/api/hotswap-deployments.ts

```diff
@@ -54,21 +56,21 @@ export async function tryHotswapDeployment(
   // create a new SDK using the CLI credentials, because the default one will not work for new-style synthesis -
   // it assumes the bootstrap deploy Role, which doesn't have permissions to update Lambda functions
   const sdk = (await sdkProvider.forEnvironment(resolvedEnv, Mode.ForWriting)).sdk;
-  // The current resources of the Stack.
-  // We need them to figure out the physical name of a resource in case it wasn't specified by the user.
-  // We fetch it lazily, to save a service call, in case all hotswapped resources have their physical names set.
-  const listStackResources = new LazyListStackResources(sdk, stackArtifact.stackName);
```

Comment:
I'd prefer we create this at the top level and pass it through instead

Reply:
We need to create a new one for nested stacks from within the `evaluate-cloudformation-template.ts` and hence thought it's better for `evaluate-cloudformation-template.ts` to manage it's own `LazyListStackResources` rather than being passed by the clients. (We can create one lazy loader for each nested stack in the entire hierarchy and pass it to the top level `evaluate-cloudformation-template.ts` but that seems overkill)